core/src/test/scala/org/apache/spark/deploy/history/EventLogFileCompactorSuite.scala

```diff
@@ -284,6 +286,20 @@ class EventLogFileCompactorSuite extends SparkFunSuite {
     }
   }
 
+  test("SPARK-38664: Support compact EventLog when there are illegal characters in the path") {
+    withTempDir { dir =>
```

Comment:
If you still want to use `withTempDir`, you need to refactor `withTempDir` to accept `namePrefix`(but this will involve many file changes). 

At present, two `tmpdirs` are created in this way... which is not necessary